src/libexpr/eval.cc

```diff
@@ -120,24 +115,32 @@ void printValue(std::ostream & str, std::set<const Value *> & active, const Valu
         str << "null";
         break;
     case tAttrs: {
-        str << "{ ";
-        for (auto & i : v.attrs->lexicographicOrder()) {
-            str << i->name << " = ";
-            printValue(str, active, *i->value);
-            str << "; ";
+        if (!v.attrs->empty() && !seen.insert(v.attrs).second)
+            str << "<REPEAT>";
```

Comment:
It seems like the `<REPEAT>` can sometimes be interpreted as a Nix diamond path (e.g. `<nixpkgs>`) -- c.f. https://github.com/NixOS/nix/issues/6243 and https://github.com/NixOS/nixpkgs/issues/163374.

(Just attempting to facilitate communication as a concerned bystander; I haven't run into this myself)